File: Quivi.Pos.Api/Dtos/MenuItem.cs
namespace Quivi.Pos.Api.Dtos
{
    public class MenuItem
    {
        public required string Id { get; init; }
        public required string Name { get; init; }
        public string? ImageUrl { get; init; }
        public decimal Price { get; init; }
        public required IEnumerable<ModifierGroup> ModifierGroups { get; init; }
    }
}

File: quivi.pos.react.app/src/hooks/api/Dtos/menuitems/MenuItem.ts
export interface MenuItem {
    readonly id: string;
    readonly name: string;
    readonly imageUrl?: string;
    readonly price: number;
    readonly modifierGroups: ModifierGroup[];
    readonly categoryIds: string[];
    readonly hasStock: boolean;
    readonly isDeleted: boolean;
}

export interface ModifierGroup {
    readonly id: string;
    readonly name: string;
    readonly minSelection: number;
    readonly maxSelection: number;
    readonly options: ModifierGroupOption[];
}

export interface ModifierGroupOption {
    readonly id: string;
    readonly menuItemId: string;
    readonly price: number;
}